packages/components/src/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,85 @@
+import { useState, ReactNode } from 'react';
+import React from 'react';
+import { type DropdownProps } from "@/dropdown/Dropdown.types";
+import { globalStyles } from "@/index.styled";
+import { DropdownContainer, DropdownLabel, DropdownChildren, DropdownItem, darkTheme } from "@/dropdown/Dropdown.styled"; 
+import chevronDown from "@/assets/chevron-down.svg"
+
+export const Dropdown = ({
+        label = "Select...", 
+        theme, 
+        showingOptions: initialShowingOptions, 
+        disabled, 
+        error, 
+        children}: DropdownProps) => {
+
+    globalStyles();
+
+    const [showingOptions, setShowingOptions] = useState(initialShowingOptions);
+    const [selectedOption, setSelectedOption] = useState('');
+
+    const toggleDropdownOptions = () => {
+        if(disabled) {
+            setShowingOptions(false);
+            return;
+        };
+        setShowingOptions(!showingOptions);
+    }
+
+    const handleDropdownKeyDown = (event: React.KeyboardEvent) => {
+        if (event.key === 'Enter' || event.key === ' ') {
+          toggleDropdownOptions();
+        }
+    }
+
+    const handleOptionClick = (option: string) => {
+        setSelectedOption(option);
+        setShowingOptions(false);
+        console.log(`Selected option: ${option}`);
+    }
+
+    const handleDropdownOptionKeyDown = (event: React.KeyboardEvent) => {
+    if (event.key === 'Enter' || event.key === ' ') {
+        if (event.currentTarget.textContent) {
+            handleOptionClick(event.currentTarget.textContent);
+        }
+    }
+}
+  
+    return (
+        <DropdownContainer
```

Comment:
When the dropdown label is set to a very long text, the options/dropdown children are not the same width as the dropdown label. It's OK to specify a fixed width and text-overflow: ellipsis here
![image](https://github.com/devsoc-unsw/ui-components/assets/112071878/ea858f2f-8d3a-4826-98e6-172cc4d6e990)